pkg/operator/ceph/controller/finalizer.go

```diff
@@ -73,25 +73,24 @@ func AddFinalizerIfNotPresent(client client.Client, obj client.Object) error {
 }
 
 // RemoveFinalizer removes a finalizer from an object
-func RemoveFinalizer(client client.Client, obj client.Object) error {
-	objectFinalizer := buildFinalizerName(obj.GetObjectKind().GroupVersionKind().Kind)
+func RemoveFinalizer(client client.Client, obj client.Object, finalizerName string) error {
```

Comment:
I think two helper methods would simplify the overall code: `RemoveFinalizer()` and `RemoveFinalizerWithName()`. The first one would automatically look up the type like before, and the latter one would take the name like you need for the new finalizers. Then you don't have to copy this line in so many places:
```
opcontroller.BuildFinalizerName(obj.GetObjectKind().GroupVersionKind().Kind)
```